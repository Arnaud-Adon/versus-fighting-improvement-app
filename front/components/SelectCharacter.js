import React from "react";
import { Text, View, FlatList } from "react-native";
import { ScrollView } from "react-native-gesture-handler";

const SelectCharacter = () => {
  return (
    <View>
      <Text testID="selectCharacterTitle">Selectionnez un personnage</Text>
      <FlatList
        testID="flatlistCharacter"
        renderItem={() => {
          <View>
            <Text>Text1</Text>
            <Text>Text2</Text>
          </View>;
        }}
      />
    </View>
  );
};

export default SelectCharacter;
