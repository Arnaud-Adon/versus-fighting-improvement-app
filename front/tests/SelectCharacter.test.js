import React from "react";
import { fireEvent, render, waitFor } from "@testing-library/react-native";
import AsyncStorage from "@react-native-community/async-storage";
import SelectCharacter from "../components/SelectCharacter";

describe("Component SelectCharacter", () => {
  test("Le composant SelectCharacter est bien rendu", async () => {
    //Arrange
    const { toJSON } = render(<SelectCharacter />);
    //Assert
    await waitFor(() => expect(toJSON()).toMatchSnapshot());
  });

  test('Contient la chaine de caractere "Selectionnez un personnage"', async () => {
    //Arrange
    const { queryByTestId } = render(<SelectCharacter />);

    //Assert
    await waitFor(() =>
      expect(queryByTestId("selectCharacterTitle").props.children).toBe(
        "Selectionnez un personnage"
      )
    );
  });

  test("Contient une Flatlist pour lister les images de personnage", async () => {
    //Arrange
    const { queryByTestId } = render(<SelectCharacter />);

    //Assert
    await waitFor(() =>
      expect(queryByTestId("flatlistCharacter")).toBeTruthy()
    );
  });

  test("Flatlist contient plusieurs items de type personnage dans sa prop renderItem", async () => {
    const { getByTestId, getByTest, queryByTestId, toJSON } = render(
      <SelectCharacter />
    );

    await waitFor(() =>
      expect(
        queryByTestId("flatlistCharacter").props.getItemCount()
      ).toBeGreaterThan(0)
    );
  });
});
