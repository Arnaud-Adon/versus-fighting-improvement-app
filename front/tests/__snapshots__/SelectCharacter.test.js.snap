// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component SelectCharacter Le composant SelectCharacter est bien rendu 1`] = `
<View>
  <Text
    testID="selectCharacterTitle"
  >
    Selectionnez un personnage
  </Text>
  <RCTScrollView
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    testID="flatlistCharacter"
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View />
  </RCTScrollView>
</View>
`;
